### ISSUE TITLE
[Guided Setup Stage 2] Add support for read-only messages (Phase 2)

### ISSUE THREAD

**Issue by francoisl** [Expensify/App#38773]
## Part of the "wave-collect – Build Stage 2 of Guided Setup" project
Main issue: https://github.com/Expensify/Expensify/issues/356685
Doc section: https://docs.google.com/document/d/10bhCv6XtzzqEZQ9tzxFwAz4lowgjOhsFqz13BMq6Iu4/edit#heading=h.9c9l6vfd8y0n
Project: https://github.com/orgs/Expensify/projects/129

### Feature Description
Use the new `permissions = ["read"]` prop on report objects to hide the composer. When the system account ("Expensify persona") is the only other participant of the report, display a static message instead of the composer instead

cc @barttom

**Comment by barttom** [comment]
Hello! I'm Bartek from Callstack, an expert contributor group, and I would like to work on this issue.

**Comment by rojiphil** [comment]
@twisterdotcom Can you please grant me access to this [document](https://docs.google.com/document/d/10bhCv6XtzzqEZQ9tzxFwAz4lowgjOhsFqz13BMq6Iu4/edit#heading=h.9c9l6vfd8y0n) mentioned in OP? I have requested now. My email is rojiphil@gmail.com

**Comment by twisterdotcom** [comment]
Ah I see we can share design docs for C+: https://stackoverflowteams.com/c/expensify/questions/11623/11624#11624.

**Comment by twisterdotcom** [comment]
Just asking in Slack whether it's best to share with individuals or with all C+ via [contributor_plus@expensify.com](mailto:contributor_plus@expensify.com): https://expensify.slack.com/archives/C01SKUP7QR0/p1711544698772059

**Comment by twisterdotcom** [comment]
Okay shared only as a commenter rojiphil@gmail.com / @rojiphil.

**Comment by twisterdotcom** [comment]
How are we doing here @barttom?

**Comment by barttom** [comment]
@twisterdotcom Basically, it's finished, but we are not able to test it without manual code changes and currently, I'm working on https://github.com/Expensify/App/issues/38771

We can test it when https://github.com/Expensify/App/issues/38769 has been finished.
Then I can implement this in. the app and I'll send this one to review with https://github.com/Expensify/App/issues/38774 - that will allow us to test this issue.

**Comment by twisterdotcom** [comment]
Nice! Thanks

**Comment by twisterdotcom** [comment]
How is this PR going? We just held on https://github.com/Expensify/App/issues/38769 right?

**Comment by rezkiy37** [comment]
> How is this PR going? We just held on #38769 right?

I prepared the [draft PR](https://github.com/Expensify/App/pull/40010). Once https://github.com/Expensify/App/issues/38769 is merged, I can start to test immediately.

**Comment by rezkiy37** [comment]
Guys, should we block those actions for read-only chats?

<img width="1379" alt="Screenshot 2024-04-22 at 12 52 08" src="https://github.com/Expensify/App/assets/57314631/ede5484a-b242-4c48-a625-f803a105e028">

cc @francoisl @twisterdotcom @mountiny

**Comment by mountiny** [comment]
Yeah I feel like we should not allow creating threads there

**Comment by mountiny** [comment]
@rezkiy37 could you bring this to the wave-collect room please?

**Comment by rezkiy37** [comment]
> @rezkiy37 could you bring this to the wave-collect room please?

https://callstack-hq.slack.com/archives/C036QM0SLJK/p1713797125727179

**Comment by trjExpensify** [comment]
As mentioned [here](https://expensify.slack.com/archives/C036QM0SLJK/p1714042407707379?thread_ts=1713799253.669479&cid=C036QM0SLJK), removing the hold now that the AuthPR for https://github.com/Expensify/App/issues/38769 is on production. Bumping the priority back to `Daily`.

**Comment by twisterdotcom** [comment]
If this is no longer held on another issue, who is this held on?

**Comment by rezkiy37** [comment]
The issue is not on hold anymore. But the PR for this issue is on hold and waiting for https://github.com/Expensify/App/pull/40678.

**Comment by rezkiy37** [comment]
Hey!
I just wanted to let you know about my vacation for next week. I will be back on 06.05.2024. Meanwhile, I asked @waterim to take over this issue. 
See you soon!

**Comment by hungvu193** [comment]
@mountiny Since I reviewed the PR, can you assign me to this issue please? Thank you 😄

**Comment by twisterdotcom** [comment]
Hey, sorry just getting to this - apologies I was out last week. I only need to pay @rojiphil and @hungvu193 via Upwork, for $250 each right? Or did @rojiphil not review an associated PR, but @hungvu193 did?

**Comment by hungvu193** [comment]
> Hey, sorry just getting to this - apologies I was out last week. I only need to pay @rojiphil and @hungvu193 via Upwork, for $250 each right? Or did @rojiphil not review an associated PR, but @hungvu193 did?

I think it's just me because I reviewed the PR.

**Comment by rojiphil** [comment]
> Hey, sorry just getting to this - apologies I was out last week. I only need to pay @rojiphil and @hungvu193 via Upwork, for $250 each right? Or did @rojiphil not review an associated PR, but @hungvu193 did?

@twisterdotcom That's correct. No payment for me as I did not review the PR.

**Comment by twisterdotcom** [comment]
Payment Summary:

- @hungvu193 paid $250 [here](https://www.upwork.com/nx/wm/offer/102386529) (C+)

**Comment by twisterdotcom** [comment]
@hungvu193 - want to propose the regression steps here?

**Comment by hungvu193** [comment]
> @hungvu193 - want to propose the regression steps here?

This is new feature so I don't think we need regression test here

**Comment by twisterdotcom** [comment]
Why would we not create regression steps for a new feature though? It is part of the normal new feature checklist: https://github.com/Expensify/App/issues/38773#issuecomment-2083836200

**Comment by hungvu193** [comment]
> Why would we not create regression steps for a new feature though? It is part of the normal new feature checklist: https://github.com/Expensify/App/issues/38773#issuecomment-2083836200

Sorry I missed it 🥲

**Comment by hungvu193** [comment]
Regression test:

1. Complete the onboarding flow.
2. Open the chat.
3. Verify that the composer is hidden.
4. Verify that a new footer component exists.
5. Open a task report.
6. Verify that the composer is hidden.
7. Verify that a new footer component exists.
8. Verify that you cannot complete or delete a task.

Do we 👍 or 👎 ?
